<a href="https://colab.research.google.com/github/shashwatshukla123/pro-114/blob/main/pro_114.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import random
import plotly.express as px
import plotly.figure_factory as ff
import statistics
import csv
import pandas as pd
import plotly.graph_objects as go
import numpy as np

In [4]:
df=pd.read_csv('main.csv')
gre_score=df["GRE Score"].tolist()
chances_of_admit=df["Chance of Admit "].tolist()
fig=px.scatter(x=gre_score,y=chances_of_admit)
fig.show()

y=mx+c 
Here we observe gre-score and chances-of-admittion are related to each other the more is GRE Score than more is Chance of Admit.
Every line follow the formula y=mx+c where m is the slope of line, c is the intersecpt of the line.
x is independent variable[GRE Score]

In [6]:
# By hit&try method we have to fid the value of m&c
m=0.01
c=-2.5
y=[]
for x in gre_score:
  y_value=m*x+c
  y.append(y_value)

# Plot the points 
fig=px.scatter(x=gre_score,y=chances_of_admit)
fig.update_layout(shapes=[
                          dict(type='line',y0=min(y),y1=max(y),x0=min(gre_score),x1=max(gre_score))
])
fig.show()

In [7]:
# we can predit Chance of Admit based on gre_score
x=600
y=m*x+c
print(f"Chance of Admittion of someone based on his gre_score {x} is {y}")

Chance of Admittion of someone based on his gre_score 600 is 3.5


*We fond the value of slope and intersect hit & tryal meathod now let's check the computer algothreum to find m and c*

In [8]:
gre_score_array=np.array(gre_score)
Chance_of_Admit_array=np.array(chances_of_admit)

# Slope and intersect using pre-built function of numpy
m,c=np.polyfit(gre_score_array,Chance_of_Admit_array,1)
print(m,c)

0.009975882025681383 -2.4360842448510547


In [9]:
gre_score_array=np.array(gre_score)
Chance_of_Admit_array=np.array(chances_of_admit)

# Slope and intersect using pre-built function of numpy
m,c=np.polyfit(gre_score_array,Chance_of_Admit_array,1)
y=[]
for x in gre_score_array:
  y_value=m*x+c
  y.append(y_value)

# Plot the points 
fig=px.scatter(x=gre_score_array,y=Chance_of_Admit_array)
fig.update_layout(shapes=[
                          dict(type='line',y0=min(y),y1=max(y),x0=min(gre_score_array),x1=max(gre_score_array))
])
fig.show()

In [10]:
# Let us predit the value of y using the new value of m and c [got from the agorithm]
x=600
y=m*x+c
print(f"Chance of Admittion of someone based on his gre_score {x} is {y}")

Chance of Admittion of someone based on his gre_score 600 is 3.5494449705577744


The chances of admit value we got from our algorithm value if 3.5494449705577744 and from ho and trial is 3.5

Linear regression is used to predit the relation between 2-varibles by appling linear equation to observed data
 there are 2 types of variable (dependent and independent)
 example:- the weight of a person is linerily related to their height that mean if we increase the height the weight of the person will also increase.

 x=>independent variable ploted along x-axis.
 
 y=>dependent variable ploted along y-axis.